### DISCUSSION TITLE
Remove after_commit_everywhere from gemspec?

### DISCUSSION THREAD

**Discussion by davekaro** [janko/sequel-activerecord_connection#32] [General]
Does it make sense to remove this from the gemspec and offer guidance instead (for example, how Sidekiq does https://github.com/sidekiq/sidekiq/blob/5054ee46fa6cea66d7acaea6e78112c895ea1a28/lib/sidekiq/transaction_aware_client.rb#L41-L45)?

I realize that in the case of Sidekiq, if you want the features provided by the after_commit_everywhere gem, it's because you are opting in to transaction aware client, whereas for sequel-activerecord_connection, it's required for core functionality. I'm just thinking for users on Rails 7.2+, it would be nice to not see this gem in the lock file, even though it's not being required or used at all.

**Comment by janko** [comment]
I agree 👍🏻 I would need to bump the major version, though, since it would be a breaking change.

I would first like to give the latest gem version a few weeks in the wild, in case someone reports an issue with the switch to the new `with_connection` API, and I would need to make another breaking change.

**Reply by janko** [reply]
I just released version 2.0.0, which requires `after_commit_everywhere` to be added to the Gemfile manually on Active Record 7.1 or older.

If you're using rodauth-rails, you will need to wait until I release a new major version there as well. I planned to make all dependencies optional there as well, so I'm glad I have an excuse for a major version bump now 🙂
